[2024-09-14T17:50:36.776+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-09-14T17:50:36.856+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: airbnb_model.train_model manual__2024-09-14T17:50:30.952450+00:00 [queued]>
[2024-09-14T17:50:36.910+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: airbnb_model.train_model manual__2024-09-14T17:50:30.952450+00:00 [queued]>
[2024-09-14T17:50:36.912+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 4
[2024-09-14T17:50:36.977+0000] {taskinstance.py:2888} INFO - Executing <Task(SparkSubmitOperator): train_model> on 2024-09-14 17:50:30.952450+00:00
[2024-09-14T17:50:37.005+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'airbnb_model', 'train_model', 'manual__2024-09-14T17:50:30.952450+00:00', '--job-id', '667', '--raw', '--subdir', 'DAGS_FOLDER/air_price_pred_dag.py', '--cfg-path', '/tmp/tmp6equ24nc']
[2024-09-14T17:50:37.010+0000] {standard_task_runner.py:105} INFO - Job 667: Subtask train_model
[2024-09-14T17:50:37.018+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=3033) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-09-14T17:50:37.020+0000] {standard_task_runner.py:72} INFO - Started process 3034 to run task
[2024-09-14T17:50:37.175+0000] {task_command.py:467} INFO - Running <TaskInstance: airbnb_model.train_model manual__2024-09-14T17:50:30.952450+00:00 [running]> on host fb744bf0dc86
[2024-09-14T17:50:37.524+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Denis' AIRFLOW_CTX_DAG_ID='airbnb_model' AIRFLOW_CTX_TASK_ID='train_model' AIRFLOW_CTX_EXECUTION_DATE='2024-09-14T17:50:30.952450+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-09-14T17:50:30.952450+00:00'
[2024-09-14T17:50:37.528+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-09-14T17:50:37.687+0000] {base.py:84} INFO - Retrieving connection 'spark_default'
[2024-09-14T17:50:37.691+0000] {spark_submit.py:403} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --name arrow-spark --verbose --deploy-mode client /opt/***/dags/air_train_model.py
[2024-09-14T17:50:44.568+0000] {spark_submit.py:579} INFO - Using properties file: null
[2024-09-14T17:50:44.983+0000] {spark_submit.py:579} INFO - Parsed arguments:
[2024-09-14T17:50:44.985+0000] {spark_submit.py:579} INFO - master                  spark://spark-master:7077
[2024-09-14T17:50:44.986+0000] {spark_submit.py:579} INFO - remote                  null
[2024-09-14T17:50:44.987+0000] {spark_submit.py:579} INFO - deployMode              client
[2024-09-14T17:50:44.988+0000] {spark_submit.py:579} INFO - executorMemory          null
[2024-09-14T17:50:44.989+0000] {spark_submit.py:579} INFO - executorCores           null
[2024-09-14T17:50:44.990+0000] {spark_submit.py:579} INFO - totalExecutorCores      null
[2024-09-14T17:50:44.991+0000] {spark_submit.py:579} INFO - propertiesFile          null
[2024-09-14T17:50:44.992+0000] {spark_submit.py:579} INFO - driverMemory            null
[2024-09-14T17:50:44.994+0000] {spark_submit.py:579} INFO - driverCores             null
[2024-09-14T17:50:44.995+0000] {spark_submit.py:579} INFO - driverExtraClassPath    null
[2024-09-14T17:50:44.997+0000] {spark_submit.py:579} INFO - driverExtraLibraryPath  null
[2024-09-14T17:50:44.998+0000] {spark_submit.py:579} INFO - driverExtraJavaOptions  null
[2024-09-14T17:50:44.999+0000] {spark_submit.py:579} INFO - supervise               false
[2024-09-14T17:50:45.000+0000] {spark_submit.py:579} INFO - queue                   null
[2024-09-14T17:50:45.001+0000] {spark_submit.py:579} INFO - numExecutors            null
[2024-09-14T17:50:45.002+0000] {spark_submit.py:579} INFO - files                   null
[2024-09-14T17:50:45.003+0000] {spark_submit.py:579} INFO - pyFiles                 null
[2024-09-14T17:50:45.004+0000] {spark_submit.py:579} INFO - archives                null
[2024-09-14T17:50:45.005+0000] {spark_submit.py:579} INFO - mainClass               null
[2024-09-14T17:50:45.007+0000] {spark_submit.py:579} INFO - primaryResource         file:/opt/***/dags/air_train_model.py
[2024-09-14T17:50:45.012+0000] {spark_submit.py:579} INFO - name                    arrow-spark
[2024-09-14T17:50:45.013+0000] {spark_submit.py:579} INFO - childArgs               []
[2024-09-14T17:50:45.015+0000] {spark_submit.py:579} INFO - jars                    null
[2024-09-14T17:50:45.016+0000] {spark_submit.py:579} INFO - packages                null
[2024-09-14T17:50:45.017+0000] {spark_submit.py:579} INFO - packagesExclusions      null
[2024-09-14T17:50:45.018+0000] {spark_submit.py:579} INFO - repositories            null
[2024-09-14T17:50:45.020+0000] {spark_submit.py:579} INFO - verbose                 true
[2024-09-14T17:50:45.021+0000] {spark_submit.py:579} INFO - 
[2024-09-14T17:50:45.022+0000] {spark_submit.py:579} INFO - Spark properties used, including those specified through
[2024-09-14T17:50:45.023+0000] {spark_submit.py:579} INFO - --conf and those from the properties file null:
[2024-09-14T17:50:45.024+0000] {spark_submit.py:579} INFO - (spark.master,spark://spark-master:7077)
[2024-09-14T17:50:45.026+0000] {spark_submit.py:579} INFO - 
[2024-09-14T17:50:45.027+0000] {spark_submit.py:579} INFO - 
[2024-09-14T17:50:46.022+0000] {spark_submit.py:579} INFO - Main class:
[2024-09-14T17:50:46.024+0000] {spark_submit.py:579} INFO - org.apache.spark.deploy.PythonRunner
[2024-09-14T17:50:46.026+0000] {spark_submit.py:579} INFO - Arguments:
[2024-09-14T17:50:46.028+0000] {spark_submit.py:579} INFO - file:/opt/***/dags/air_train_model.py
[2024-09-14T17:50:46.030+0000] {spark_submit.py:579} INFO - null
[2024-09-14T17:50:46.036+0000] {spark_submit.py:579} INFO - Spark config:
[2024-09-14T17:50:46.037+0000] {spark_submit.py:579} INFO - (spark.app.name,arrow-spark)
[2024-09-14T17:50:46.039+0000] {spark_submit.py:579} INFO - (spark.app.submitTime,1726336245964)
[2024-09-14T17:50:46.040+0000] {spark_submit.py:579} INFO - (spark.master,spark://spark-master:7077)
[2024-09-14T17:50:46.042+0000] {spark_submit.py:579} INFO - (spark.submit.deployMode,client)
[2024-09-14T17:50:46.044+0000] {spark_submit.py:579} INFO - (spark.submit.pyFiles,)
[2024-09-14T17:50:46.046+0000] {spark_submit.py:579} INFO - Classpath elements:
[2024-09-14T17:50:46.048+0000] {spark_submit.py:579} INFO - 
[2024-09-14T17:50:46.049+0000] {spark_submit.py:579} INFO - 
[2024-09-14T17:50:46.051+0000] {spark_submit.py:579} INFO - 
[2024-09-14T17:50:51.415+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:51 INFO SparkContext: Running Spark version 3.5.2
[2024-09-14T17:50:51.427+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:51 INFO SparkContext: OS info Linux, 5.15.153.1-microsoft-standard-WSL2, amd64
[2024-09-14T17:50:51.429+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:51 INFO SparkContext: Java version 17.0.12
[2024-09-14T17:50:51.696+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:51 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-09-14T17:50:52.066+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:52 INFO ResourceUtils: ==============================================================
[2024-09-14T17:50:52.068+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:52 INFO ResourceUtils: No custom resources configured for spark.driver.
[2024-09-14T17:50:52.071+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:52 INFO ResourceUtils: ==============================================================
[2024-09-14T17:50:52.072+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:52 INFO SparkContext: Submitted application: Airbnb_train_model
[2024-09-14T17:50:52.170+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:52 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2024-09-14T17:50:52.203+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:52 INFO ResourceProfile: Limiting resource is cpu
[2024-09-14T17:50:52.206+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:52 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2024-09-14T17:50:52.508+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:52 INFO SecurityManager: Changing view acls to: ***
[2024-09-14T17:50:52.512+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:52 INFO SecurityManager: Changing modify acls to: ***
[2024-09-14T17:50:52.514+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:52 INFO SecurityManager: Changing view acls groups to:
[2024-09-14T17:50:52.516+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:52 INFO SecurityManager: Changing modify acls groups to:
[2024-09-14T17:50:52.518+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:52 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2024-09-14T17:50:53.769+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:53 INFO Utils: Successfully started service 'sparkDriver' on port 38603.
[2024-09-14T17:50:54.017+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:54 INFO SparkEnv: Registering MapOutputTracker
[2024-09-14T17:50:54.315+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:54 INFO SparkEnv: Registering BlockManagerMaster
[2024-09-14T17:50:54.375+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:54 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2024-09-14T17:50:54.378+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:54 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2024-09-14T17:50:54.397+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:54 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2024-09-14T17:50:54.480+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:54 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-61968ac4-9e76-45aa-ab9f-0e2db3220cae
[2024-09-14T17:50:54.539+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:54 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2024-09-14T17:50:54.612+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:54 INFO SparkEnv: Registering OutputCommitCoordinator
[2024-09-14T17:50:55.075+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:55 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2024-09-14T17:50:55.331+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:55 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2024-09-14T17:50:55.990+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:55 INFO Executor: Starting executor ID driver on host fb744bf0dc86
[2024-09-14T17:50:56.011+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:55 INFO Executor: OS info Linux, 5.15.153.1-microsoft-standard-WSL2, amd64
[2024-09-14T17:50:56.013+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:55 INFO Executor: Java version 17.0.12
[2024-09-14T17:50:56.030+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:56 INFO Executor: Starting executor with user classpath (userClassPathFirst = false): ''
[2024-09-14T17:50:56.033+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:56 INFO Executor: Created or updated repl class loader org.apache.spark.util.MutableURLClassLoader@2b2d567f for default.
[2024-09-14T17:50:56.135+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:56 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 44901.
[2024-09-14T17:50:56.137+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:56 INFO NettyBlockTransferService: Server created on fb744bf0dc86:44901
[2024-09-14T17:50:56.144+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:56 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2024-09-14T17:50:56.171+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:56 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, fb744bf0dc86, 44901, None)
[2024-09-14T17:50:56.184+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:56 INFO BlockManagerMasterEndpoint: Registering block manager fb744bf0dc86:44901 with 434.4 MiB RAM, BlockManagerId(driver, fb744bf0dc86, 44901, None)
[2024-09-14T17:50:56.193+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:56 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, fb744bf0dc86, 44901, None)
[2024-09-14T17:50:56.196+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:56 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, fb744bf0dc86, 44901, None)
[2024-09-14T17:50:58.649+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:58 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2024-09-14T17:50:58.690+0000] {spark_submit.py:579} INFO - 24/09/14 17:50:58 INFO SharedState: Warehouse path is 'file:/user/hive/warehouse'.
[2024-09-14T17:51:10.442+0000] {job.py:239} ERROR - Job heartbeat failed with error
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 3371, in _wrap_pool_connect
    return fn()
           ^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/pool/base.py", line 327, in connect
    return _ConnectionFairy._checkout(self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/pool/base.py", line 894, in _checkout
    fairy = _ConnectionRecord.checkout(pool)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/pool/base.py", line 493, in checkout
    rec = pool._do_get()
          ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/pool/impl.py", line 256, in _do_get
    return self._create_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/pool/base.py", line 273, in _create_connection
    return _ConnectionRecord(self)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/pool/base.py", line 388, in __init__
    self.__connect()
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/pool/base.py", line 690, in __connect
    with util.safe_reraise():
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    compat.raise_(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/pool/base.py", line 686, in __connect
    self.dbapi_connection = connection = pool._invoke_creator(self)
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/create.py", line 574, in connect
    return dialect.connect(*cargs, **cparams)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/default.py", line 598, in connect
    return self.dbapi.connect(*cargs, **cparams)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.OperationalError: could not translate host name "postgres" to address: Temporary failure in name resolution


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/jobs/job.py", line 233, in heartbeat
    heartbeat_callback(session)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/jobs/local_task_job_runner.py", line 284, in heartbeat_callback
    self.task_instance.refresh_from_db()
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 2358, in refresh_from_db
    _refresh_from_db(task_instance=self, session=session, lock_for_update=lock_for_update)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 872, in _refresh_from_db
    ti = TaskInstance.get_task_instance(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/api_internal/internal_api_call.py", line 139, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 94, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 2344, in get_task_instance
    return query.one_or_none()
           ^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/orm/query.py", line 2850, in one_or_none
    return self._iter().one_or_none()
           ^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/orm/query.py", line 2916, in _iter
    result = self.session.execute(
             ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/orm/session.py", line 1716, in execute
    conn = self._connection_for_bind(bind)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/orm/session.py", line 1555, in _connection_for_bind
    return self._transaction._connection_for_bind(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/orm/session.py", line 750, in _connection_for_bind
    conn = bind.connect()
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/future/engine.py", line 412, in connect
    return super(Engine, self).connect()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 3325, in connect
    return self._connection_cls(self, close_with_result=close_with_result)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 96, in __init__
    else engine.raw_connection()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 3404, in raw_connection
    return self._wrap_pool_connect(self.pool.connect, _connection)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 3374, in _wrap_pool_connect
    Connection._handle_dbapi_exception_noconnection(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 2208, in _handle_dbapi_exception_noconnection
    util.raise_(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 3371, in _wrap_pool_connect
    return fn()
           ^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/pool/base.py", line 327, in connect
    return _ConnectionFairy._checkout(self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/pool/base.py", line 894, in _checkout
    fairy = _ConnectionRecord.checkout(pool)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/pool/base.py", line 493, in checkout
    rec = pool._do_get()
          ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/pool/impl.py", line 256, in _do_get
    return self._create_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/pool/base.py", line 273, in _create_connection
    return _ConnectionRecord(self)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/pool/base.py", line 388, in __init__
    self.__connect()
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/pool/base.py", line 690, in __connect
    with util.safe_reraise():
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    compat.raise_(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/pool/base.py", line 686, in __connect
    self.dbapi_connection = connection = pool._invoke_creator(self)
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/create.py", line 574, in connect
    return dialect.connect(*cargs, **cparams)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/engine/default.py", line 598, in connect
    return self.dbapi.connect(*cargs, **cparams)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) could not translate host name "postgres" to address: Temporary failure in name resolution

(Background on this error at: https://sqlalche.me/e/14/e3q8)
[2024-09-14T17:51:10.800+0000] {job.py:245} ERROR - Job heartbeat failed with error. Scheduler may go into unhealthy state
[2024-09-14T17:51:12.051+0000] {spark_submit.py:579} INFO - Traceback (most recent call last):
[2024-09-14T17:51:12.102+0000] {spark_submit.py:579} INFO - File "/opt/***/dags/air_train_model.py", line 21, in <module>
[2024-09-14T17:51:12.105+0000] {spark_submit.py:579} INFO - df = spark.read.option("header", "true").option("inferSchema", "true").csv(csv_path)
[2024-09-14T17:51:12.109+0000] {spark_submit.py:579} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-09-14T17:51:12.110+0000] {spark_submit.py:579} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/sql/readwriter.py", line 740, in csv
[2024-09-14T17:51:12.116+0000] {spark_submit.py:579} INFO - return self._df(self._jreader.csv(self._spark._sc._jvm.PythonUtils.toSeq(path)))
[2024-09-14T17:51:12.135+0000] {spark_submit.py:579} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-09-14T17:51:12.137+0000] {spark_submit.py:579} INFO - File "/home/***/.local/lib/python3.12/site-packages/py4j/java_gateway.py", line 1322, in __call__
[2024-09-14T17:51:12.139+0000] {spark_submit.py:579} INFO - return_value = get_return_value(
[2024-09-14T17:51:12.140+0000] {spark_submit.py:579} INFO - ^^^^^^^^^^^^^^^^^
[2024-09-14T17:51:12.143+0000] {spark_submit.py:579} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/errors/exceptions/captured.py", line 185, in deco
[2024-09-14T17:51:12.152+0000] {spark_submit.py:579} INFO - raise converted from None
[2024-09-14T17:51:13.250+0000] {spark_submit.py:579} INFO - pyspark.errors.exceptions.captured.AnalysisException: [PATH_NOT_FOUND] Path does not exist: file:/user/hive/warehouse/processed_airbnb_data_csv.
[2024-09-14T17:51:14.333+0000] {spark_submit.py:579} INFO - 24/09/14 17:51:14 INFO SparkContext: Invoking stop() from shutdown hook
[2024-09-14T17:51:14.351+0000] {spark_submit.py:579} INFO - 24/09/14 17:51:14 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2024-09-14T17:51:14.398+0000] {spark_submit.py:579} INFO - 24/09/14 17:51:14 INFO SparkUI: Stopped Spark web UI at http://fb744bf0dc86:4040
[2024-09-14T17:51:14.508+0000] {spark_submit.py:579} INFO - 24/09/14 17:51:14 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2024-09-14T17:51:14.954+0000] {spark_submit.py:579} INFO - 24/09/14 17:51:14 INFO MemoryStore: MemoryStore cleared
[2024-09-14T17:51:14.985+0000] {spark_submit.py:579} INFO - 24/09/14 17:51:14 INFO BlockManager: BlockManager stopped
[2024-09-14T17:51:14.997+0000] {spark_submit.py:579} INFO - 24/09/14 17:51:14 INFO BlockManagerMaster: BlockManagerMaster stopped
[2024-09-14T17:51:15.019+0000] {spark_submit.py:579} INFO - 24/09/14 17:51:15 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2024-09-14T17:51:15.368+0000] {spark_submit.py:579} INFO - 24/09/14 17:51:15 INFO SparkContext: Successfully stopped SparkContext
[2024-09-14T17:51:15.398+0000] {spark_submit.py:579} INFO - 24/09/14 17:51:15 INFO ShutdownHookManager: Shutdown hook called
[2024-09-14T17:51:15.399+0000] {spark_submit.py:579} INFO - 24/09/14 17:51:15 INFO ShutdownHookManager: Deleting directory /tmp/spark-271ac074-da21-4e75-8a34-45681d2bb8bc
[2024-09-14T17:51:15.522+0000] {spark_submit.py:579} INFO - 24/09/14 17:51:15 INFO ShutdownHookManager: Deleting directory /tmp/spark-25a0374f-5e48-43c1-b3c7-7672c12934c5
[2024-09-14T17:51:15.545+0000] {spark_submit.py:579} INFO - 24/09/14 17:51:15 INFO ShutdownHookManager: Deleting directory /tmp/spark-25a0374f-5e48-43c1-b3c7-7672c12934c5/pyspark-334561ae-9709-47a7-84a4-a252509d871c
[2024-09-14T17:51:16.108+0000] {job.py:229} INFO - Heartbeat recovered after 15.66 seconds
[2024-09-14T17:51:16.117+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 505, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --name arrow-spark --verbose --deploy-mode client /opt/***/dags/air_train_model.py. Error code is: 1.
[2024-09-14T17:51:16.186+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=airbnb_model, task_id=train_model, run_id=manual__2024-09-14T17:50:30.952450+00:00, execution_date=20240914T175030, start_date=20240914T175036, end_date=20240914T175116
[2024-09-14T17:51:16.261+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-09-14T17:51:16.263+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 667 for task train_model (Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --name arrow-spark --verbose --deploy-mode client /opt/***/dags/air_train_model.py. Error code is: 1.; 3034)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 505, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark-master:7077 --name arrow-spark --verbose --deploy-mode client /opt/***/dags/air_train_model.py. Error code is: 1.
[2024-09-14T17:51:16.359+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-09-14T17:51:16.440+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-09-14T17:51:16.458+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
